Add sum filter documentation (#1732)
Shopify/liquid: _filters/sum.md

---
title: sum
description: Liquid filter that sums all items in an array.
---

Sums all items in an array.

If a string is passed as an argument, it sums the property values.

> In this example, assume the object `collection.products` contains a list of products, and products has a `quantity` property. Using assign with the sum filter creates a variable that contains the total amount of quantities in all products from the collection.

Reviewer: ```suggestion
In this example, assume the object `collection.products` contains a list of products, and each `product` object has a `quantity` property. Using `assign` with the `sum` filter creates a variable that contains the total amount of quantities for all products in the collection.
```

---
title: sum
description: Liquid filter that sums all items in an array.
---

Sums all items in an array.

<p class="code-label">Input</p>
```liquid
{%- raw -%}
<!-- if array = [1,2,3] -->
{{ array | sum }}
{% endraw %}
```

<p class="code-label">Output</p>
```text
6
```

 If a string is passed as an argument, it sums the property values.

<p class="code-label">Input</p>
```liquid
{%- raw -%}
<!-- if array = [
  {quantity: 1, weight: 2},
  {quantity: 2},

>   quantity: 3}

Reviewer 2: ?
```suggestion
  {quantity: 3}
```

> ]

Reviewer: Since liquid does not have object literals, it might be beneficial to use an intention-revealing array of objects (e.g., `collection.products`) like they do in the `map` documentation (`"In this example, assume the object site.pages contains all the metadata for a website..."`)
